test/test_project.py

```diff
@@ -421,6 +422,38 @@ def test_module_use_everything_reexport(copy_fortran_file):
     }
 
 
+def test_use_within_interface(copy_fortran_file):
+    data = """\
+    module module1
+      implicit none
+      private
+      interface
+        subroutine routine_1()
+          use module2, only: routine_2
+        end subroutine routine_1
+      end interface
+      public :: routine
+    end module module1
+
+    module module2
+      implicit none
+      private
+      public :: routine_2
+    contains
+      subroutine routine_2(i)
+        integer,intent(in) :: i
+        write(*,*) i
+      end subroutine routine_2
+    end module module2
+    """
+
+    settings = copy_fortran_file(data)
+    project = create_project(settings)
+    assert set(
+        project.modules[0].all_procs["routine_1"].procedure.all_procs.keys()
+    ) == {"routine_1", "routine_2"}
```

Comment:
This is a little surprising! Should `routine_1` appear in `all_procs` for itself? Is this the previous behaviour?

Reply:
Yes, this is the previous behavior, and yeah, I think this may be incorrect. I've removed that part from the test and now only check what I'm looking for in case this gets fixed in the future.

Reply:
Very odd. There might be a good reason for this behaviour, but it's not immediately obvious. Might make sense for recursive procedures?